server/client.go

```diff
@@ -1506,17 +1506,27 @@ func (c *client) processConnect(arg []byte) error {
 	account := c.opts.Account
 	accountNew := c.opts.AccountNew
 
-	ncs := c.String()
-	if c.opts.Name != "" {
-		ncs = fmt.Sprintf("%s - name: %s", ncs, c.opts.Name)
-	}
-	if c.opts.Lang != "" {
-		ncs = fmt.Sprintf("%s - lang: %s", ncs, c.opts.Lang)
-	}
-	if c.opts.Version != "" {
-		ncs = fmt.Sprintf("%s - version: %s", ncs, c.opts.Version)
+	if c.kind == CLIENT {
+		var ncs string
+		if c.opts.Version != "" {
+			ncs = fmt.Sprintf("v%s", c.opts.Version)
+		}
+		if c.opts.Lang != "" {
+			if c.opts.Version == _EMPTY_ && c.opts.Name == _EMPTY_ {
+				ncs = c.opts.Lang
+			} else {
+				ncs = fmt.Sprintf("%s:%s", ncs, c.opts.Lang)
+			}
+		}
+		if c.opts.Name != "" {
+			if c.opts.Version == _EMPTY_ {
```

Comment:
In that order, you would need here to check that both Version and Lang are empty to set `ncs = c.opts.Name`.